server/src/main/java/org/elasticsearch/index/query/GeoDistanceQueryBuilder.java

```diff
@@ -244,7 +244,7 @@ protected Query doToQuery(SearchExecutionContext context) throws IOException {
 
         final GeoShapeQueryable geoShapeQueryable = (GeoShapeQueryable) fieldType;
         final Circle circle = new Circle(center.lon(), center.lat(), this.distance);
-        return geoShapeQueryable.geoShapeQuery(circle, fieldType.name(), SpatialStrategy.RECURSIVE, ShapeRelation.INTERSECTS, context);
+        return geoShapeQueryable.geoShapeQuery(context, fieldType.name(), SpatialStrategy.RECURSIVE, ShapeRelation.INTERSECTS, circle);
```

Comment:
```suggestion
        return geoShapeQueryable.geoShapeQuery(context, fieldType.name(), ShapeRelation.INTERSECTS, circle);
```

Reply:
Needed for compatibility with `LegacyGeoShapeFieldMapper`